File: src/CorporateMessengerWithAgile/Application/Dto/EmployeeDto.cs
namespace Application.Dto
{
    public class EmployeeDto : BaseDto
    {
        public Guid CompanyId { get; set; }
        public Guid PositionInCompanyId { get; set; }
        public Guid UserId { get; set; }
        public IReadOnlyList<Guid> TeamMemberIds { get; set; } = null!;
    }
}

File: src/CorporateMessengerWithAgile/ReactApp/reactapp.client/src/models/entity/EmployeeDto.ts
// src/models/entity/EmployeeDto.ts
import { z } from 'zod';
import { BaseDtoSchema } from './BaseDto';
import { GuidSchema } from '../Guid';

export const EmployeeDtoSchema = BaseDtoSchema.extend({
    companyId: GuidSchema,
    positionInCompanyId: GuidSchema,
    userId: GuidSchema,
    teamMemberIds: z.array(GuidSchema).optional(),
});

export type EmployeeDto = z.infer<typeof EmployeeDtoSchema>;
